boa/src/syntax/ast/node/return_smt.rs

```diff
@@ -0,0 +1,63 @@
+use super::Node;
+use gc::{Finalize, Trace};
+use std::fmt;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+/// The `return` statement ends function execution and specifies a value to be returned to the
+/// function caller.
+///
+/// Syntax: `return [expression];`
+///
+/// `expression`:
+///  > The expression whose value is to be returned. If omitted, `undefined` is returned
+///  > instead.
+///
+/// When a `return` statement is used in a function body, the execution of the function is
+/// stopped. If specified, a given value is returned to the function caller.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///  - [MDN documentation][mdn]
+///
+/// [spec]: https://tc39.es/ecma262/#prod-ReturnStatement
+/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/return
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(transparent))]
+#[derive(Clone, Debug, Trace, Finalize, PartialEq)]
+pub struct Return {
+    expr: Option<Box<Node>>,
+}
+
+impl Return {
+    pub fn expr(&self) -> &Option<Box<Node>> {
+        &self.expr
+    }
```

Comment:
This should also return Option<&Node>:
```suggestion
    pub fn expr(&self) -> Option<&Node> {
        self.expr.as_ref().map(Box::as_ref)
    }
```